drivers/sensor/wsen/wsen_hids_2525020210002/wsen_hids_2525020210002.c

```diff
@@ -0,0 +1,252 @@
+	data->sensor_heater = (hids_2525020210002_heater_t)heater->val1;
+
+	return 0;
+}
+
+static int hids_2525020210002_attr_set(const struct device *dev, enum sensor_channel chan,
+				       enum sensor_attribute attr, const struct sensor_value *val)
+{
+
+	if (chan != SENSOR_CHAN_ALL) {
+		LOG_WRN("attr_set() is not supported on channel %d.", chan);
+		return -ENOTSUP;
+	}
+
+	switch ((int)attr) {
+	case SENSOR_ATTR_WSEN_HIDS_2525020210002_PRECISION:
+		return hids_2525020210002_precision_set(dev, val);
+	case SENSOR_ATTR_WSEN_HIDS_2525020210002_HEATER:
+		return hids_2525020210002_heater_set(dev, val);
+	default:
+		LOG_ERR("Operation not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int hids_2525020210002_attr_get(const struct device *dev, enum sensor_channel chan,
+				       enum sensor_attribute attr, struct sensor_value *val)
+{
+	struct hids_2525020210002_data *data = dev->data;
+
+	if (chan != SENSOR_CHAN_ALL) {
+		LOG_WRN("attr_get() is not supported on channel %d.", chan);
+		return -ENOTSUP;
+	}
+
+	if (val == NULL) {
+		LOG_WRN("address of passed value is NULL.");
+		return -EFAULT;
+	}
+
+	switch ((int)attr) {
+	case SENSOR_ATTR_WSEN_HIDS_2525020210002_PRECISION:
+		val->val1 = data->sensor_precision;
+		val->val2 = 0;
+		break;
+	case SENSOR_ATTR_WSEN_HIDS_2525020210002_HEATER:
+		val->val1 = data->sensor_heater;
+		val->val2 = 0;
+		break;
+	default:
+		LOG_ERR("Operation not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api hids_2525020210002_driver_api = {
+	.attr_set = hids_2525020210002_attr_set,
+	.attr_get = hids_2525020210002_attr_get,
+	.sample_fetch = hids_2525020210002_sample_fetch,
+	.channel_get = hids_2525020210002_channel_get,
+};
+
+static int hids_2525020210002_init(const struct device *dev)
+{
+	const struct hids_2525020210002_config *const config = dev->config;
+	struct hids_2525020210002_data *data = dev->data;
+	struct sensor_value precision, heater;
+
+	/* Initialize WE sensor interface */
+	HIDS_Get_Default_Interface(&data->sensor_interface);
+	data->sensor_interface.interfaceType = WE_i2c;
+	if (!i2c_is_ready_dt(&config->bus_cfg.i2c)) {
+		LOG_ERR("I2C bus device not ready");
+		return -ENODEV;
+	}
+	data->sensor_interface.handle = (void *)&config->bus_cfg.i2c;
+
+	/* First communication test - check device ID */
+	if (HIDS_Sensor_Init(&data->sensor_interface) != WE_SUCCESS) {
+		LOG_ERR("Failed to read device ID.");
+		return -EIO;
+	}
+
+	precision.val1 = config->precision;
+	precision.val2 = 0;
+
+	if (hids_2525020210002_precision_set(dev, &precision) < 0) {
+		LOG_ERR("Failed to set precision configuration.");
+		return -EIO;
+	}
+
+	heater.val1 = config->heater;
+	heater.val2 = 0;
+
+	if (hids_2525020210002_heater_set(dev, &heater) < 0) {
+		LOG_ERR("Failed to set heater option.");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "HIDS-2525020210002 driver enabled without any devices"
+#endif
```

Comment:
Is this redundant if CMake depends on the following?

``` Kconfig
config WSEN_HIDS_2525020210002
	bool "WSEN-HIDS-2525020210002 humidity sensor"
	default y
	depends on DT_HAS_WE_WSEN_HIDS_2525020210002_ENABLED
```

Reply:
this is also redundant and has been removed